app_integrations/apps/aliyun.py

```diff
@@ -0,0 +1,148 @@
+"""
+Copyright 2018-present, Airbnb Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+import json
+import re
+
+from aliyunsdkcore.client import AcsClient
+from aliyunsdkcore.acs_exception.exceptions import ClientException
+from aliyunsdkcore.acs_exception.exceptions import ServerException
+from aliyunsdkactiontrail.request.v20171204 import LookupEventsRequest
+
+from app_integrations import LOGGER
+from app_integrations.apps.app_base import StreamAlertApp, AppIntegration
+
+@StreamAlertApp
+class AliyunApp(AppIntegration):
+    """An app that collects events from the ActionTrail feature of Aliyun.
+
+    Results are paginated, with a NextToken field provided that is used in subsequent queries.
+    The result of a successful api call is json whose outermost schema is
+
+    {
+      "EndTime": <end of the time range of events>,
+      "NextToken": <the token to use to request the next page of data>,
+      "RequestId": <the ID of the request>,
+      "StartTime": <start of the time range of events>,
+      "Events": [entries],
+    }
+
+    If there are no more events in the queried range, the 'NextToken' element is not present.
+
+    Resource:
+        https://www.alibabacloud.com/help/doc-detail/28849.htm
+    """
+
+    _MAX_RESULTS = 50
+
+    def __init__(self, config):
+        super(AliyunApp, self).__init__(config)
+        self._next_token = None
+
+    @classmethod
+    def _type(cls):
+        return 'events'
+
+    @classmethod
+    def service(cls):
+        return 'aliyun'
+
+
+    @classmethod
+    def date_formatter(cls):
+        """Return a format string for a date, ie: 2014-05-26T12:00:00Z
+
+        This format is consisten with the format used by the Aliyun API:
+            https://www.alibabacloud.com/help/doc-detail/28849.htm
+        """
+        return '%Y-%m-%dT%H:%M:%SZ'
+
+    def _gather_logs(self):
+        auth = self._config.auth
+        client = AcsClient(auth['access_key_id'], auth['access_key_secret'], auth['region_id'])
```

Comment:
the `_gather_logs` method could be called a bunch of times in an execution - therefore, would it make sense to have the `AcsClient` created upon `__init__` and cached as an instance property?